src/sql/base/browser/ui/table/treeGrid.ts

```diff
@@ -0,0 +1,183 @@
+		}
+
+		this._grid.onClick.subscribe((e, data) => {
+			this.setRowExpandedState(data.row);
+			return false;
+		});
+
+		// The events returned by grid are Jquery events. These events can be handled by returning false which executes preventDefault and stopPropagation
+		this._grid.onKeyDown.subscribe((e, data) => {
+			const keyboardEvent = (<any>e).originalEvent;
+			if (keyboardEvent instanceof KeyboardEvent) {
+				let event = new StandardKeyboardEvent(keyboardEvent);
+				if (event.keyCode === KeyCode.Enter) {
+					// toggle the collapsed state of the row
+					this.setRowExpandedState(data.row);
+					return false;
+				} else if (event.keyCode === KeyCode.LeftArrow) {
+					// Left arrow on first cell of the expanded row collapses it
+					if (data.cell === 0) {
+						this.setRowExpandedState(data.row, false); // Collapsing state
+						return false;
+					}
+				} else if (event.keyCode === KeyCode.RightArrow) {
+					// Right arrow on last cell of the collapsed row expands it.
+					if (data.cell === (this._grid.getColumns().length - 1)) {
+						this.setRowExpandedState(data.row, true);
+						return false;
+					}
+				}
+			}
+			return true;
+		});
+
+		this._grid.onRendered.subscribe((e, data) => {
+			const visibleRows = this._grid.getViewport();
+			for (let i = visibleRows.top; i <= visibleRows.bottom; i++) {
+				const rowData = this._data.getItem(i);
+				// Getting the row div that corresponds to the data row
+				const rowElement = this._tableContainer.querySelector(`div [role="row"][aria-rowindex="${(i + 1)}"]`);
+				// If the row element is found in the dom, we are setting the required aria attributes for it.
+				if (rowElement) {
+					if (rowData.expanded !== undefined) {
+						rowElement.ariaExpanded = rowData.expanded;
+					} else {
+						rowElement.removeAttribute('aria-expanded');
+					}
+					if (rowData.setSize !== undefined) {
+						rowElement.ariaSetSize = rowData.setSize;
+					} else {
+						rowElement.removeAttribute('aria-setsize');
+					}
+					if (rowData.posInSet !== undefined) {
+						rowElement.ariaPosInSet = rowData.posInSet;
+					} else {
+						rowElement.removeAttribute('aria-posinset');
+					}
+					if (rowData.level !== undefined) {
+						rowElement.ariaLevel = rowData.level;
+					}
+				}
+			}
+			return false;
+		});
+	}
+
+	override setData(data: Array<T>): void;
+	override setData(data: TableDataView<T>): void;
+	override setData(data: AsyncDataProvider<T>): void;
+	override setData(data: Array<T> | TableDataView<T> | AsyncDataProvider<T>): void {
+		if (data instanceof TableDataView || data instanceof AsyncDataProvider) {
+			this._data = data;
+		} else {
+			this._data = new TableDataView<T>(data, undefined, undefined, defaultTreeGridFilter);
+		}
+		this.addTreeGridDataAttributes(this._data);
+		this._grid.setData(this._data, true);
+		this._data.filter(this._grid.getColumns());
+	}
+
+	public override set columns(columns: Slick.Column<T>[]) {
+		if (columns[0]) {
```

Comment:
This is duplicated with the code in the constructor. Consolidate this into one - preferably by updating the base constructor to call this property setter instead.